src/commands/init/index.ts

```diff
@@ -79,30 +142,56 @@ export default class Init extends Command {
         return projectConfig;
     }
 
-    async ensurePackagesAreDownloaded(projectConfig: ProjectConfig, force: boolean, verbose: boolean) {
-        for (const packageConfig of projectConfig.getPackages()) {
-            const packageVersions = await packageConfig.getPackageVersions();
-            const packageVersion = resolveVersionIdentifier(packageVersions, packageConfig.version);
+    private async _resolveVersion(packageConfig: PackageConfig, verbose: boolean) {
+        const packageVersions = await packageConfig.getPackageVersions();
+        const packageVersion = resolveVersionIdentifier(packageVersions, packageConfig.version);
 
-            if (verbose) {
-                this.log(`... Resolved '${packageConfig.getPackageName()}:${packageConfig.version}' to version: '${packageVersion}'`);
-            }
+        if (verbose) {
+            this.log(`... Resolved '${packageConfig.getPackageName()}:${packageConfig.version}' to version: '${packageVersion}'`);
+        }
 
-            packageConfig.setPackageVersion(packageVersion);
+        packageConfig.setPackageVersion(packageVersion);
+    }
 
-            if (!force && packageConfig.isPackageInstalled()) {
-                this.log(`... '${packageConfig.getPackageName()}:${packageConfig.version}' already installed.`);
-                continue;
+    private async _modifyProjectConfig(
+        requestedPackageConfig: PackageConfig,
+        projectConfig: ProjectConfig,
+        force: boolean,
+        verbose: boolean,
+    ) {
+        const packageConfigs = projectConfig
+            .getPackages()
+            .filter((config) => config.getPackageName() === requestedPackageConfig.getPackageName());
+
+        const isPackageExisting = packageConfigs.length >= 1;
+        if (!isPackageExisting) {
+            this.log(`... Package '${requestedPackageConfig.getPackageName()}:${requestedPackageConfig.version}' added to .velocitas.json`);
+            projectConfig.addPackage(requestedPackageConfig);
+
+            packageConfigs.push(requestedPackageConfig);
+        } else {
+            this.log(`... Package '${requestedPackageConfig.getPackageName()}' exists in .velocitas.json`);
+            const existingPackageConfig = packageConfigs.at(0);
+            if (existingPackageConfig?.version !== requestedPackageConfig.version) {
+                this.log(`... Package '${requestedPackageConfig.getPackageName()}' updated to version ${requestedPackageConfig.version}`);
+                existingPackageConfig?.setPackageVersion(requestedPackageConfig.version);
             }
+        }
+    }
 
-            this.log(`... Downloading package: '${packageConfig.getPackageName()}:${packageConfig.version}'`);
-            await packageConfig.downloadPackageVersion(verbose);
+    private async _ensurePackageIsDownloaded(packageConfig: PackageConfig, force: boolean, verbose: boolean) {
```

Comment:
this can stay almost like before:
```
private async _ensurePackagesAreDownloaded(packageConfigs: PackageConfig[], force: boolean, verbose: boolean) {
        for (const packageConfig of packageConfigs) {
            await this._resolveVersion(packageConfig, verbose);
```

Reply:
Done, however this means in case of '_handleSinglePackageInit' that we are basically resolving the version twice, once when creating the requestedPackageConfig, and once during download of the packages.

Reply:
yes thats true, do this for now and we check later